### ISSUE TITLE
[BUG] `ERR_REQUIRE_ESM` when using config file

### ISSUE THREAD

**Issue by opack** [chenasraf/simple-scaffold#66]
#### Describe the bug

Hello! I stumbled on a bug when trying to use the config file. I get `ERR_REQUIRE_ESM`:
```shell
$ npx simple-scaffold@latest -c scaffold.config.js -k service SomeService
Loading config from scaffold.config.js with key service
Loading config from file scaffold.config.js
node:internal/process/promises:288
            triggerUncaughtException(err, true /* fromPromise */);
            ^

Error [ERR_REQUIRE_ESM]: require() of ES Module C:\Users\micro\Projects\my-project\scaffold.config.js from C:\Users\micro\Projects\my-project\node_modules\simple-scaffold\config.js not supported.
scaffold.config.js is treated as an ES module file as it is a .js file whose nearest parent package.json contains "type": "module" which declares all .js files in that package scope as ES modules.
Instead rename scaffold.config.js to end in .cjs, change the requiring code to use dynamic import() which is available in all CommonJS modules, or change "type": "module" to "type": "commonjs" in C:\Users\micro\Projects\my-project\package.json to treat all .js files as CommonJS (using .mjs for all ES mo
dules instead).

    at C:\Users\micro\Projects\my-project\node_modules\simple-scaffold\config.js:119:104 {
  code: 'ERR_REQUIRE_ESM'
}

Node.js v18.17.1
```
Works fine when using only the command-line with `-t` and `-o` options.

#### How To Reproduce

1. Prepare templates:

   ```text
   // scaffold.config.js
   module.exports = {
	 service: {
	   templates: ['tools/templates/service'],
	   output: '.'
	 }
   }
   ```

2. Run with args/config:

   ```shell
   npx simple-scaffold@latest -c scaffold.config.js -k service SomeService
   ```

#### Expected behavior\*\*

Files generated without aby error, similar to when I run `npx simple-scaffold@latest -t tools/templates/service -o . SomeService`

#### Logs
Not applicable, and anyway adding this parameter does not add anything to the console.

#### Desktop (please complete the following information):

- OS: Windows
- OS Version: 10 Pro 22H2
- Node.js: 18.17.1
- Simple Scaffold Version 1.8.0

**Comment by chenasraf** [comment]
Hi @opack, sorry for the trouble!

Can you try renaming it to .cjs and see what happens?

In the meantime I will be investigating this and try to find a solution

**Comment by opack** [comment]
Hi! Thanks for the quick answer 😉 
When I rename to cjs, I have this error:
```shell
$ npx simple-scaffold@latest -c scaffold.config.js -k service SomeService
Loading config from scaffold.config.js with key service
Loading config from file scaffold.config.js                                                                         
node:internal/modules/cjs/loader:1077                                                                               
  const err = new Error(message);                                                                                   
              ^                                                                                                     
                                                                                                                    
Error: Cannot find module 'C:\Users\micro\Projects\my-project\scaffold.config.js'                               
Require stack:                                                                                                      
- C:\Users\micro\AppData\Local\npm-cache\_npx\42a4b0a410ce2116\node_modules\simple-scaffold\config.js               
- C:\Users\micro\AppData\Local\npm-cache\_npx\42a4b0a410ce2116\node_modules\simple-scaffold\file.js
- C:\Users\micro\AppData\Local\npm-cache\_npx\42a4b0a410ce2116\node_modules\simple-scaffold\scaffold.js
- C:\Users\micro\AppData\Local\npm-cache\_npx\42a4b0a410ce2116\node_modules\simple-scaffold\cmd.js
    at Module._resolveFilename (node:internal/modules/cjs/loader:1077:15)
    at Module._load (node:internal/modules/cjs/loader:922:27)
    at Module.require (node:internal/modules/cjs/loader:1143:19)
    at require (node:internal/modules/cjs/helpers:121:18)
    at C:\Users\micro\AppData\Local\npm-cache\_npx\42a4b0a410ce2116\node_modules\simple-scaffold\config.js:119:104 {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    'C:\\Users\\micro\\AppData\\Local\\npm-cache\\_npx\\42a4b0a410ce2116\\node_modules\\simple-scaffold\\config.js',
    'C:\\Users\\micro\\AppData\\Local\\npm-cache\\_npx\\42a4b0a410ce2116\\node_modules\\simple-scaffold\\file.js',
    'C:\\Users\\micro\\AppData\\Local\\npm-cache\\_npx\\42a4b0a410ce2116\\node_modules\\simple-scaffold\\scaffold.js',
    'C:\\Users\\micro\\AppData\\Local\\npm-cache\\_npx\\42a4b0a410ce2116\\node_modules\\simple-scaffold\\cmd.js'
  ]
}

Node.js v18.17.1
```

**Comment by chenasraf** [comment]
You should also rename the file in the CLI flag accordingly - 
```
npx simple-scaffold@latest -c scaffold.config.cjs -k service SomeService
```

**Comment by opack** [comment]
Oh my god, I'm so stupid, I'm ashamed: obviously it works now that I use the correct file name 😅

Thanks a lot! 🙏 You can close this issue if you will, or leave it open if you want to find another solution. Maybe just updating the doc would be ok 😉

**Comment by chenasraf** [comment]
I'm just glad you managed to get it working :) 

I won't close this for now, I want to see if there is a solution I can give, or at the very least, manage to properly detect the error and maybe give hints. Worst case, I'll only update the docs.

Thanks for letting me know! :PRAY:

**Comment by chenasraf** [comment]
Fixed in `2.0.0-pre.1`. Also, you can now omit the `-c` flag, if your filename is one of the [pre-defined searched files](https://chenasraf.github.io/simple-scaffold/docs/usage/configuration_files#using-a-config-file).
